jdbc/sqlserver/src/main/java/com/google/cloud/sql/sqlserver/SocketFactory.java

```diff
@@ -36,8 +37,16 @@ public class SocketFactory extends javax.net.SocketFactory {
    * Implements the {@link SocketFactory} constructor, which can be used to create authenticated
    * connections to a Cloud SQL instance.
    */
-  public SocketFactory(String instanceName) {
-    this.props.setProperty(CoreSocketFactory.CLOUD_SQL_INSTANCE_PROPERTY, instanceName);
+  public SocketFactory(String socketFactoryConstructorArg) {
+    String[] s = socketFactoryConstructorArg.split("\\?");
+    this.props.setProperty(CoreSocketFactory.CLOUD_SQL_INSTANCE_PROPERTY, s[0]);
+    if (s.length == 2) {
+      String[] queryParams = s[1].split("&");
```

Comment:
URLs have funny rules about escaping characters etc. I wonder if you could use some method on java.util.URI to parse this instead of rolling your own.

Reply:
Are we using this as an actual URL though? I assumed we were just adapting the syntax because it was familiar.

Reply:
My feeling is that if we're going to say "like URL syntax" in the docs, then we need to support full URL syntax. Otherwise someone is going to want to pass a query parameter value with a `=` character in it and get stuck. 

After doing a bit of research, I think your code can stay mostly the same with String.split(), but use java.net.URLDecoder.decode(url, StandardCharsets.UTF_8.name()) on the parameter keys and values.